### PR TITLE
Cloud Resources

### PR DESCRIPTION
**Pull request by dehume**
Fixes https://github.com/MaterializeInc/pulumi-materialize/issues/90

Outline for updating the Pulumi provider for TF Cloud resources:

1. Set the version of the Terraform provider in `provider/go.mod` (assuming it will be `0.5.0`) and run `go mod tidy` within the `provider` directory
2. Confirm that all the provider attributes and resources are added are expected
3. Generate the code `make tfgen` (from the root)
4. Ensure everything is running as expected by running the tests for the Python SDK
```
docker compose -f examples/docker-compose.yml up -d --build

export MZ_ENDPOINT=http://localhost:3000
export MZ_CLOUD_ENDPOINT=http://localhost:3001
export MZ_PASSWORD=mzp_1b2a3c4d5e6f7a8b9c0d1e2f3a4b5c6d7e8f9a0b1c2d3e4f5a6b7c8d9e0f1a2b
export MZ_SSLMODE=disable

make test
```
5. If everything looks good, merge and cut the release. Maybe consider this being version `0.1.0`

### CODE REVIEW

File: examples/docker-compose.yml [hunk 1]
```diff
@@ -18,3 +18,15 @@ services:
       - 6877:6877
       - 6878:6878
     healthcheck: {test: curl -f localhost:6878/api/readyz, interval: 1s, start_period: 35s}
+
+  frontegg:
```
Comment (dehume) [inline examples/docker-compose.yml:22]: All the mocks and additional services for the docker compose are from [the](https://github.com/MaterializeInc/terraform-provider-materialize/pull/351). There is probably a better way to mirror this.

File: provider/go.mod [hunk 2]
```diff
@@ -5,7 +5,7 @@ go 1.18
 replace github.com/hashicorp/terraform-plugin-sdk/v2 => github.com/pulumi/terraform-plugin-sdk/v2 v2.0.0-20230912190043-e6d96b3b8f7e
 
 require (
-	github.com/MaterializeInc/terraform-provider-materialize v0.3.4
+	github.com/MaterializeInc/terraform-provider-materialize v0.5.0
```
Comment (dehume) [inline provider/go.mod:8]: Will be whatever the TF version for the cloud resources is.

File: provider/resources.go [hunk 3]
```diff
@@ -65,26 +65,27 @@ func Provider() tfbridge.ProviderInfo {
...
 			"password": {
 				Default: &tfbridge.DefaultInfo{EnvVars: []string{"MZ_PASSWORD"}},
 			},
-			"port": {
-				Default: &tfbridge.DefaultInfo{EnvVars: []string{"MZ_PORT"}},
-			},
 			"database": {
 				Default: &tfbridge.DefaultInfo{EnvVars: []string{"MZ_DATABASE"}},
 			},
+			"endpoint": {
```
Comment (dehume) [inline provider/resources.go:74]: Update the provider level attributes. Specifically setting them so environment variable values can be set.

File: provider/resources.go [hunk 4]
```diff
@@ -65,26 +65,27 @@ func Provider() tfbridge.ProviderInfo {
...
+			},
+			"default_region": {
+				Default: &tfbridge.DefaultInfo{EnvVars: []string{"MZ_DEFAULT_REGION"}},
+			},
 			"sslmode": {
 				Default: &tfbridge.DefaultInfo{EnvVars: []string{"MZ_SSLMODE"}},
 			},
 		},
 		Resources: map[string]*tfbridge.ResourceInfo{
+			"materialize_app_password":                         {Tok: tfbridge.MakeResource(mainPkg, mainMod, "AppPassword")},
```
Comment (dehume) [inline provider/resources.go:88]: Add any new Cloud resources to `Resources`. If there were datasources they would be added to `DataSources`.

**Review by bobbyiliev — APPROVED** [review]

**Pull request merged**